main.rb

```diff
@@ -0,0 +1,98 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    i = 0
+    size = self.size
+    while i < size
+      is_a?(Range) ? yield(min + i) : yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    idx = 0
+    my_each do |elem|
+      yield(elem, idx)
+      idx += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    if is_a?(Array)
+      selected = []
+      my_each { |ele| selected << ele if yield ele }
+    else
+      selected = {}
+      my_each { |key, value| selected[key] = value if yield(key, value) }
+    end
+    selected
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+  def my_all?(arg = nil, &prc)
+    return true if !block_given? && arg.nil? && include?(nil) == false && include?(false) == false
+    return false unless block_given? || !arg.nil?
+
+    if block_given?
+      my_each { |elem| return false if prc.call(elem) == false }
+    elsif arg.class == Regexp
+      my_each { |elem| return false if arg.match(elem).nil? }
+    elsif arg.class <= Numeric || arg.class <= String
+      my_each { |elem| return false if elem != arg }
+    else
+      my_each { |elem| return false if (elem.is_a? arg) == false }
+    end
+    true
+  end
+
+  def my_any?(arg = nil, &prc)
+    return true if !block_given? && arg.nil? && my_each { |elem| return true if elem == true } && empty? == false
+    return false unless block_given? || !arg.nil?
+
+    my_each do |ele|
+      return true if prc.call(ele) == true
+    end
+    false
+  end
+
+  def my_none?(arg = nil, &prc)
+    !my_any?(arg, &prc)
+  end
```

Comment:
#### my_none?

- [ ] I see you get my_none? from my any? and I think that's actually a good practice so when you solve the problems in your my_any this method should work properly and that's why I won't list the failing cases here.